spaceship/lib/spaceship/tunes/device_type.rb

```diff
@@ -1,39 +1,15 @@
+require_relative "display_family"
+
 module Spaceship
   module Tunes
     # identifiers of devices that App Store Connect accepts screenshots for
     class DeviceType
```

Comment:
In general we want to deprecate and not use this class any more, correct? 
If so, we could maybe add some output or a comment?
Does fastlane itself still use it or do we only leave it for external users (via `spaceship`)?

Reply:
`DeviceType#exists?` is used twice in Spaceship and once in Deliver and `DeviceType#types` appears to no longer be used (other than for `exists?`) 

I could introduce something like the following to `DisplayFamily` to replace the current usages:

```diff
- exists = Spaceship::Tunes::DeviceType.exists?(screenshot.device_type)
+ exists = !Spaceship::Tunes::DisplayFamily.for_name(screenshot.device_type.to_sym).nil?
```

Open to suggestions to make that look nicer if you have any 👍

Reply:
> do we only leave it for external users (via spaceship)?

So I have no way to answer this.. I just had a quick search of the project and it looks like we have a `UI.deprecated` so I guess that I could introduce something like the following:

```ruby
def self.types
  UI.deprecated("Spaceship::Tunes::DeviceType has been deprecated. Use the new Spaceship::Tunes::DisplayFamily class instead")
  return DisplayFamily.all.map(&:name)
end
```

I'm just slightly refining the methods on `DisplayFamily` so that it fits in a bit better with the way they are used in the project.. Will push it shortly

Reply:
I think we don't use `UI` in spaceship, as it should also be usable outside of the context of fastlane as a standalone library.

So probably the easier way will be to just add a comment that explains that we don't use this any more (if we actually do and you find a good way to replace its usage what you posted in https://github.com/fastlane/fastlane/pull/14574#discussion_r275161079 looks fine to me).

Reply:
So yeah I just realised that 🙈 

After some googling, I saw a suggestion to use `warn` to print a message and I just committed that but I could just change it to a comment if you still think that it's better?

I've gone through the uses in the main project and replaced them with DisplayFamily so in theory we could just delete this file if we wanted as well

Reply:
> I've gone through the uses in the main project and replaced them with DisplayFamily so in theory we could just delete this file if we wanted as well

Yeah, the problem is that people might use spaceship in isolation as a library, and depend on this file and its methods - that's why I think we should keep it around - otherwise we have a breaking change.

Reply:
Ah yeah this makes sense... Deprecating seems like a good approach for the time being then. Are you happy with the `warn` message or should I change it to a comment?

Reply:
I am perfectly happy. The question is if ruby and other people knowing more about proper ruby etiquette are ;) Will try to get some other people form the team to look at it.